crates/transport-http/src/lib.rs

```diff
@@ -37,19 +36,28 @@ use url::Url;
 pub struct HttpConnect<T> {
     /// The URL to connect to.
     url: Url,
-    _pd: PhantomData<T>,
+    transport: Option<T>,
 }
 
 impl<T> HttpConnect<T> {
     /// Create a new [`HttpConnect`] with the given URL.
     pub const fn new(url: Url) -> Self {
-        Self { url, _pd: PhantomData }
+        Self { url, transport: None }
+    }
+
+    pub const fn with_transport(transport: T, url: Url) -> Self {
```

Comment:
why do we need to add this?

Reply:
ah I see, that's to have `TransportConnect` impl for any instance of `HyperTransport`

I am not sure we should have that. I believe that point of `TransportConnect` is in knowing how to instantiate transport. If we require user to provide it with a ready transport instance, then user can just directly configure this instance instead of relying on `TransportConnect`